### PR TITLE
Upgrade to sbt 1.4.4

### PR DESCRIPTION
**Pull request by smarter**
Theo following projects failed to compile: minitest, scopt, squants and
stdlib213:
- Scopt and squants still do not support Scala 3 upstream, I decided to
  drop them because the alternatives (staying with an old fork and
  cherry-picking changes, or continuously rebasing our fork on top of
  upstream) requires too much on-going maintenance work, we can add them
  back when they officially support Scala 3.
- Minitest was upgraded to the latest upstream commit (I made a PR
  upstream to support 3.0.0-M2), at the same time I enabled running the
  tests (both JVM and JS) of minitest instead of just compiling it.
- The new minitest required a new scalacheck, upgrading scalacheck broke
  other projects in the community build which depended on an old version
  of it. I fixed the issue by using `dependencyOverrides` to force all
  projects to use the same version of scalacheck, we can use this
  mechanism for other projects upgrade in the future.
- stdlib213 was rebased on the latest upstream commit, I opened a PR
  with the necessary changes so we don't need to keep a fork anymore:
  https://github.com/scala/scala/pull/9345

Some of our sbt scripted tests started failing after upgrading sbt, I
traced this to an unintentional behavior change in sbt which now uses
always run scripted tests in batch
mode (https://github.com/sbt/sbt/issues/5913#issuecomment-716003195).
The next two commits add batch modes support to our scripted tests by
porting fixes made in sbt itself when it switched its own tests to batch
modes a while ago. The good news is that the batch mode makes the
scripted tests much faster.

### CODE REVIEW

**Review by griggt — COMMENTED** [review]
Is there a reason for the change to dotty-cps-async?  It looks like the HEAD of the submodule has been pointed away from master which was updated earlier today by the merge of #10486.

It seems that much of the test logging in the community build is now suppressed: the log for part a has gone from ~15k to ~6k lines, and part b from ~13k to ~2k lines.  I presume this is expected/intentional.

I'm pleasantly surprised by the performance improvement from the use of batch mode for the sbt scripted tests. On my local machine, running time was reduced from 35 to 8 minutes!

I was also a bit surprised not to see [test_sbt] in the body of the pull request message.  ;-)

**Comment by smarter** [comment]
> Is there a reason for the change to dotty-cps-async?

Nope, looks like I messed up a rebase, will fix, thanks for the catch.

> I was also a bit surprised not to see [test_sbt] in the body of the pull request message. ;-)

Yeah I tried it on a personal fork so I didn't have to but it would have been a good use for it. :)

**Review by griggt — APPROVED** [review]

File: sbt-dotty/sbt-test/source-dependencies/subproject-dependency-b/build.sbt [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+import complete.DefaultParsers._
 
-InputKey[Unit]("check-number-of-compiler-iterations") <<= inputTask { (argTask: TaskKey[Seq[String]]) =>
-  (argTask, compile in Compile in use) map { (args: Seq[String], a: sbt.inc.Analysis) =>
-    assert(args.size == 1)
-    val expectedIterationsNumber = args(0).toInt
-    assert(a.compilations.allCompilations.size == expectedIterationsNumber, 
-      "a.compilations.allCompilations.size = %d (expected %d)".format(a.compilations.allCompilations.size, expectedIterationsNumber))
+// Reset compiler iterations, necessary because tests run in batch mode
+val recordPreviousIterations = taskKey[Unit]("Record previous iterations.")
```
Comment (anatoliykmetyuk) [inline sbt-dotty/sbt-test/source-dependencies/subproject-dependency-b/build.sbt:5]: This task is duplicated throughout the codebase. It's fine to get it in to unblock other stuff dependent on it, but we should create an issue to DRY it out in the future.

**Review by anatoliykmetyuk — APPROVED** [review]
Otherwise LGTM.

File: sbt-dotty/sbt-test/source-dependencies/subproject-dependency-b/build.sbt (hunk 1 above)
Comment (anatoliykmetyuk) [reply sbt-dotty/sbt-test/source-dependencies/subproject-dependency-b/build.sbt:5]: Maybe a compiler plugin would help here?

Comment (smarter) [reply sbt-dotty/sbt-test/source-dependencies/subproject-dependency-b/build.sbt:5]: The problem is to figure out how to inject a compiler plugin in the scripted tests, one way would be to have it as a separate project that we publish before we run the scripted tests, but that's pretty heavy-weight.

**Comment by SethTisue** [comment]
fyi @eed3si9n @eatkins, pretty cool to see both Scala 2 and Scala 3 on latest sbt again (cf https://github.com/scala/scala/pull/9340, https://github.com/scala/scala/pull/9202)

**Pull request merged**